h5py/_hl/group.py

```diff
@@ -284,6 +284,9 @@ def __getitem__(self, name):
             oid = h5r.dereference(name, self.id)
             if oid is None:
                 raise ValueError("Invalid HDF5 object reference")
+        elif not isinstance(name, (bytes, str)):
+            raise TypeError("Accessing a group is done with bytes or str, "
+                            " not {}".format(type(name)))
         else:
             oid = h5o.open(self.id, self._e(name), lapl=self._lapl)
```

Comment:
```suggestion
        elif isinstance(name, (bytes, str)):
            oid = h5o.open(self.id, self._e(name), lapl=self._lapl)
        else:
            raise TypeError("Accessing a group is done with bytes or str, "
                            " not {}".format(type(name)))
```

This is logically exactly the same, so it's not important, but I think it's slightly neater to group together the cases it can handle, and have the error in a final `else`. E.g. if we one day wanted to handle another type, we could just drop another `elif` in.

Reply:
Thanks!